bee-ternary/src/num_conversions.rs

```diff
@@ -0,0 +1,194 @@
+use crate::raw::{
+    RawEncoding,
+    RawEncodingBuf,
+};
+use std::convert::TryFrom;
+
+pub use crate::{
+    t1b1::{
+        T1B1Buf,
+        T1B1,
+    },
+    trit::{
+        Btrit,
+        ShiftTernary,
+        Trit,
+        Utrit,
+    },
+    TritBuf,
+};
+
+use std::convert::{
+    From,
+    Into,
+};
+
+use std::panic;
+
+const RADIX: u8 = 3;
+const NUM_TRITS_FOR_NUM_TYPE: usize = 27;
+const MAX_TRITS_IN_I64: usize = (0.63 * 64 as f32) as usize + 1; // (log2/log3)*64
+
+fn min_trits(value: u64) -> usize {
+    let mut num = 1;
+    let mut vp: u64 = 1;
+
+    let mut value = value;
+    let mut num = 1;
+    while value > vp {
+        vp = vp * RADIX as u64 + 1;
+        num = num + 1;
+    }
+
+    num
+}
+
+//TODO - generalize over all encodings
+//Impl copied from:
+// [https://github.com/iotaledger/iota_common/blob/master/common/trinary/trit_long.c#L62]
+impl From<i64> for TritBuf<T1B1Buf> {
+    fn from(value: i64) -> Self {
+        let negative = value < 0;
+        // Edge case where value == i64::MIN. In this case,
+        // "abs" cannot return a value greater than i64::MAX
+        // Which it should since the range is not symmetrical
+        // so we "force" the (u64) value explicitly
+        let mut value_abs = match value {
+            std::i64::MIN => (std::i64::MAX as u64 + 1),
+            _ => value.abs() as u64,
+        };
+
+        let size = min_trits(value_abs);
+        let mut buf = Self::zeros(size);
+
+        let mut last_pos = 0;
+        for pos in 0..size {
+            if value_abs == 0 {
+                break;
+            }
+            let mut curr_trit = ((value_abs + 1) % (RADIX as u64)) as i8 - 1;
+            if negative {
+                curr_trit = -curr_trit;
+            }
+
+            buf.set(pos, Btrit::try_from(curr_trit).unwrap());
```

Comment:
Can this fail? If not, we should make a note why not. If yes, then this implementation should be done in terms of `TryFrom`.

Reply:
Let a comment:
This can not fail because curr_trit is "some_value % 3 - 1" which is always within range